src/modules/post/repository.rs

```diff
@@ -33,32 +37,51 @@ impl PostRepository {
         Self { database }
     }
 
-    pub async fn insert(&self, user: User, content: &str) -> Result<Post> {
-        let conn = self.database.conn_pool.get()?;
-        let dto = InsertPostTableRow {
-            content: content.to_string(),
-            user_id: user.id.clone(),
-        };
-        let row = diesel::insert_into(posts::table)
-            .values(dto)
-            .get_result::<PostsTableRow>(&conn)?;
-        let post = Post {
-            id: row.id,
-            content: row.content,
-            author: user,
-            created_at: row.created_at,
-            updated_at: row.updated_at,
-        };
-
-        Ok(post)
+    pub async fn find_by_author(&self, author_id: &Uuid) -> Result<Vec<PostsTableRow>> {
```

Comment:
The struct `PostsTableRow` is meant to handle database interactions (is a DTO). But we must keep methods consistent. Please return `Post` instead of `PostsTableRow` here.

```suggestion
    pub async fn find_by_author(&self, author_id: &Uuid) -> Result<Vec<Post>> {
```